presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchPageSourceProvider.java

```diff
@@ -53,10 +58,33 @@ public ConnectorPageSource createPageSource(
         requireNonNull(split, "split is null");
         requireNonNull(table, "table is null");
 
-        return new ElasticsearchPageSource(
+        ElasticsearchTableHandle elasticsearchTable = (ElasticsearchTableHandle) table;
+        ElasticsearchSplit elasticsearchSplit = (ElasticsearchSplit) split;
+
+        if (columns.isEmpty()) {
+            long count = client.count(
+                    elasticsearchSplit.getIndex(),
+                    elasticsearchSplit.getShard(),
+                    buildSearchQuery(session, elasticsearchTable.getConstraint().transform(ElasticsearchColumnHandle.class::cast), elasticsearchTable.getQuery()));
```

Comment:
`ScanQueryPageSource` is going to repeat the same logic.
Maybe do it here once, and pass to `client.count` or `ScanQueryPageSource`?

Reply:
That's less ideal. I consider that an implementation detail of the specific page source. This branch of the `if` is really a "CountQueryPageSource", but I thought it overkill to create a dedicated class for it.

I moved this whole block into a separate dedicated method.